test/e2e/tests/confirmations/navigation.spec.ts

```diff
@@ -0,0 +1,216 @@
+
+        await driver.clickElement('[data-testid="previous-page"]');
+
+        // Verify Sign Typed Data v3 confirmation is displayed
+        assert.ok(await origin);
+        assert.ok(await message);
+      },
+    );
+  });
+
+  it('initiates multiple signatures and rejects all', async function () {
+    await withRedesignConfirmationFixtures(
+      this.test?.fullTitle(),
+      async ({ driver }: { driver: Driver }) => {
+        await unlockWallet(driver);
+        await openDapp(driver);
+        await queueSignatures(driver);
+        await driver.delay(regularDelayMs);
+
+        await driver.clickElement('[data-testid="confirm-nav__reject-all"]');
+
+        await driver.waitUntilXWindowHandles(2);
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+
+        await verifyRejectionResults(driver, '#signTypedDataResult');
+        await verifyRejectionResults(driver, '#signTypedDataV3Result');
+        await verifyRejectionResults(driver, '#signTypedDataV4Result');
+      },
+    );
+  });
+});
+
+async function verifyRejectionResults(driver: Driver, verifyResultId: string) {
+  const rejectionResult = await driver.findElement(verifyResultId);
+  assert.equal(
+    await rejectionResult.getText(),
+    'Error: User rejected the request.',
+  );
+}
+
+async function verifySignedTypeV3Confirmation(
+  driver: Driver,
+  origin: WebElement,
+) {
+  const fromAddress = driver.findElement({
+    css: '.name__value',
+    text: '0xCD2a3...DD826',
+  });
+  const toAddress = driver.findElement({
+    css: '.name__value',
+    text: '0xbBbBB...bBBbB',
+  });
+  const contents = driver.findElement({ text: 'Hello, Bob!' });
+
+  assert.ok(await origin, 'origin');
+  assert.ok(await fromAddress, 'fromAddress');
+  assert.ok(await toAddress, 'toAddress');
+  assert.ok(await contents, 'contents');
+}
+
+async function verifySignedTypeV4Confirmation(
+  driver: Driver,
+  origin: WebElement,
+) {
+  verifySignedTypeV3Confirmation(driver, origin);
+  const attachment = driver.findElement({ text: '0x' });
+  assert.ok(await attachment, 'attachment');
+}
+
+async function queueSignatures(driver: Driver) {
+  await driver.clickElement('#signTypedData');
+  await driver.waitUntilXWindowHandles(3);
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+
+  await driver.clickElement('#signTypedDataV3');
+  await driver.waitUntilXWindowHandles(3);
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+
+  await driver.waitUntilXWindowHandles(3);
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+
+  await driver.clickElement('#signTypedDataV4');
+  await driver.waitUntilXWindowHandles(3);
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+}
+
+async function queueSignaturesAndTransactions(driver: Driver) {
+  await driver.clickElement('#signTypedData');
+  await driver.waitUntilXWindowHandles(3);
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
```

Comment:
could we skip this one since we aren't doing anything on this dialog yet? same comment for the other `WINDOW_TITLES.Dialog` below

Reply:
The next step verifies on the dialog, I can move it from this function to the main test both will end up achieving the same result. Wdyt?